src/runtime/scheduler.cc

```diff
@@ -14,10 +16,15 @@ bool is_local( shared_ptr<IRuntime> rt )
   return rt->get_info()->link_speed == numeric_limits<double>::max();
 }
 
-void get( shared_ptr<IRuntime> worker, Handle<Relation> job, Relater& rt )
+void get( shared_ptr<IRuntime> worker, Handle<AnyDataType> job, Relater& rt )
 {
   if ( is_local( worker ) ) {
-    worker->get( job );
+    job.visit<void>( overload {
+      [&]( Handle<Literal> ) {},
+      [&]( Handle<Named> h ) { worker->get( h ); },
+      [&]( Handle<AnyTree> h ) { worker->get( h ); },
+      [&]( Handle<Relation> h ) { worker->get( h ); },
```

Comment:
```suggestion
      [&]( Handle<Literal> ) {},
      [&]( auto h ) { worker->get( h ); },
```

I think according to the spec type inference should be able to figure this out, although idk if that's true in practice.